### ISSUE TITLE
focusWrap support for `useToolbar`

### ISSUE THREAD

**Issue by ritz078** [adobe/react-spectrum#5494]
### Provide a general summary of the feature here

It will be great if the focus wraps inside the toolbar. 

### 🤔 Expected Behavior?

There should be an option to enable focus wrap

### 😯 Current Behavior

There is no option to enable wrapping.

### 💁 Possible Solution

I believe the change should not be big since focus manager already supports wrap

### 🔦 Context

This feature is very useful if you have a lot of items in the toolbar.

**Comment by snowystinger** [comment]
This is in the pattern https://www.w3.org/WAI/ARIA/apg/patterns/toolbar/ however, it's optional and it comes with some caveats for controls which may need to use Left/Right arrows within a horizontal toolbar.

We haven't implemented Home/End for the toolbar either. Maybe that would be better, though elements we eventually want to support need those as well, such as Select.

Possibly exposing them as a prop instead could help with this, however, a user would need to know when it was appropriate to use each one.

All this to say, I think this needs a bit of discussion.

**Comment by ritz078** [comment]
> it's optional and it comes with some caveats for controls which may need to use Left/Right arrows within a horizontal toolbar.

> We haven't implemented Home/End for the toolbar either. Maybe that would be better, though elements we eventually want to support need those as well, such as Select.

I believe irrespective of this feature, we will run into this issue. For example, in case of vertical toolbar and a menu, should the down arrow move the focus to the next item or open the menu?

**Comment by snowystinger** [comment]
Agreed, and we haven't figured out how that interaction will work yet either.
The caveat*, though, which I referred to does "address" this. I use quotes because I don't think it addresses it well, however, if it's the expectation, then we may not want to deviate by much. I started this as a conversation with our Accessibility team during implementation, but do not have an answer yet.

Caveat* from toolbar pattern page
> Avoid including controls whose operation requires the pair of arrow keys used for toolbar navigation. If unavoidable, include only one such control and make it the last element in the toolbar. For example, in a horizontal toolbar, a textbox could be included as the last element.
